pkg/testing/command/command.go

```diff
@@ -0,0 +1,188 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package command
+
+import (
+	"bytes"
+	"errors"
+	"os/exec"
+	"syscall"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type Command struct {
+	// Name of the command to be run, used to give information.
+	Name string
+
+	// ValidateOutput is a function used to verify the output. It must make the test fail in
+	// case of error.
+	ValidateOutput func(t *testing.T, output string)
+
+	// StartAndStop indicates this command should first be started then stopped.
+	// It corresponds to gadget like execsnoop which wait user to type Ctrl^C.
+	StartAndStop bool
+
+	// started indicates this command was started.
+	// It is only used by command which have StartAndStop set.
+	started bool
+
+	// command is a Cmd object used when we want to start the command, then other
+	// do stuff and wait for its completion.
+	Cmd *exec.Cmd
+
+	// stdout contains command standard output when started using Startcommand().
+	stdout bytes.Buffer
+
+	// stderr contains command standard output when started using Startcommand().
+	stderr bytes.Buffer
+}
+
+func (c *Command) IsStartAndStop() bool {
+	return c.StartAndStop
+}
+
+func (c *Command) Running() bool {
+	return c.started
+}
+
+// initExecCmd configures c.Cmd to store the stdout and stderr in c.stdout and c.stderr so that we
+// can use them on c.verifyOutput().
+func (c *Command) initExecCmd() {
+	c.Cmd.Stdout = &c.stdout
+	c.Cmd.Stderr = &c.stderr
+
+	// To be able to kill the process of /bin/sh and its child (the process of
+	// c.Cmd), we need to send the termination signal to their process group ID
+	// (PGID). However, child processes get the same PGID as their parents by
+	// default, so in order to avoid killing also the integration tests process,
+	// we set the fields Setpgid and Pgid of syscall.SysProcAttr before
+	// executing /bin/sh. Doing so, the PGID of /bin/sh (and its children)
+	// will be set to its process ID, see:
+	// https://cs.opensource.google/go/go/+/refs/tags/go1.17.8:src/syscall/exec_linux.go;l=32-34.
+	c.Cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true, Pgid: 0}
+}
+
+// verifyOutput verifies if the stdout match with the expected regular expression and the expected
+// string. If it doesn't, verifyOutput makes the test fail.
+func (c *Command) verifyOutput(t *testing.T) {
+	output := c.stdout.String()
+
+	if c.ValidateOutput != nil {
+		c.ValidateOutput(t, output)
```

Comment:
nit. Since after your last changes the `output` variable is not needed anymore outside the `if` condition 

```suggestion
		c.ValidateOutput(t, c.stdout.String())
```